drivers/include/rn2xx3.h

```diff
@@ -0,0 +1,686 @@
+                                        *   response */
+    char resp_buf[RN2XX3_MAX_BUF];     /**< command response data buffer */
+    uint16_t resp_size;                /**< counter for received char in response */
+    uint8_t resp_done;                 /**< check if response has completed */
+
+    /* buffer for RX messages */
+    char rx_tmp_buf[2];                /**< Temporary RX buffer used to convert
+                                        *   2 hex characters in one byte on the
+                                        *   fly. */
+    uint8_t rx_buf[RN2XX3_RX_MAX_BUF]; /**< RX data buffer */
+    uint16_t rx_size;                  /**< counter for received char in RX */
+
+    /* timers */
+    xtimer_t sleep_timer;              /**< Timer used to count module sleep time */
+    uint32_t sleep;                    /**< module sleep duration */
+} rn2xx3_t;
+
+/**
+ * @brief   Prepares the given RN2XX3 device
+ *
+ * @param[out] dev          RN2XX3 device to initialize
+ * @param[in]  params       parameters for device initialization
+ */
+void rn2xx3_setup(rn2xx3_t *dev, const rn2xx3_params_t *params);
+
+/**
+ * @brief   Initializes the RN2XX3 device
+ *
+ * @param[in] dev           RN2XX3 device descriptor
+ *
+ * @return                  RN2XX3_OK on success
+ * @return                  -ENXIO if UART initialization failed
+ * @return                  RN2XX3_TIMEOUT if UART communcation failed
+ */
+int rn2xx3_init(rn2xx3_t *dev);
+
+/**
+ * @brief   Restarts the RN2XX2 device
+ *
+ *          dev->resp_buf contains the module name and version string.
+ *
+ * @param[in] dev           RN2XX3 device descriptor
+ *
+ * @return                  RN2XX3_OK on success
+ * @return                  RN2XX3_TIMEOUT if UART communcation failed
+ */
+int rn2xx3_sys_reset(rn2xx3_t *dev);
+
+/**
+ * @brief   Resets the module’s configuration data and user EEPPROM to factory
+ *          default values and restarts the module
+ *
+ *          dev->resp_buf contains the module name and version string.
+ *
+ * @param[in] dev           RN2XX3 device descriptor
+ *
+ * @return                  RN2XX3_OK on success
+ * @return                  RN2XX3_TIMEOUT if UART communcation failed
+ */
+int rn2xx3_sys_factory_reset(rn2xx3_t *dev);
+
+/**
+ * @brief   Puts the RN2XX2 device in sleep mode
+ *
+ * @param[in] dev           RN2XX3 device descriptor
+ *
+ * @return                  RN2XX3_OK on success
+ * @return                  RN2XX3_TIMEOUT if UART communcation failed
+ */
+ int rn2xx3_sys_sleep(rn2xx3_t *dev);
+
+/**
+ * @brief   Initializes the RN2XX3 device MAC layer
+ *
+ * @param[in] dev           RN2XX3 device descriptor
+ *
+ * @return                  RN2XX3_OK on success
+ * @return                  RN2XX3_TIMEOUT if UART communcation failed
+ */
+int rn2xx3_mac_init(rn2xx3_t *dev);
+
+/**
+ * @brief   Writes a command to the RN2XX3 device
+ *
+ *          The module will immediately reply with a meaningful message if
```

Comment:
here and many other places: indent off. Long doc indents same as ```@``` (one space after star)